hicommonwealth/commonwealth
# Added open magic wallet functionality

## Link to Issue
Closes: #5309 

## Description of Changes
- Added open magic wallet functionality for accounts made through email and Google

**NOTE: I made these changes based off of the [Figma design](https://www.figma.com/design/rBTow8pp1mvKyWmEM8z3aP/Magic-Wallet?node-id=2-1789&node-type=frame&t=07MJEJ7PgtSBjCs4-0)s. I tried desperately to embed the widget in the account creation modal as was shown in the original designs, but unfortunately widget's don't work that way. Sabina and I decided to simply replace it with an image and change the copy telling the user where they can access their magic wallet**

## Test Plan
- Sign in with email
- confirm you see the arrow-square-out icon, and when you hover over it a tooltip reads "Open wallet"
- click the icon and confirm you see the Magic wallet widget
- go to a community with Stake and click "Buy Stake"
- confirm you see the "Add funds" button
- click it and confirm you see the Magic widget
- click the profile dropdown and confirm you see "Open wallet" with the arrow-square-out icon
- click either the text or the icon and confirm that you see the Magic widget
- in the Profile dropdown click "My community stake" 
- on My Community Stake page confirm you see the Open wallet button  
- click the button and confirm you see the Magic widget.
- Sign in with Google and go through all of the above steps
- Sign in with any other wallet/method and confirm that you do not see any of the above, as they are set to only show when a user has a Magic wallet. 
<img width="334" alt="Screenshot 2024-11-25 at 10 12 43 AM" src="https://github.com/user-attachments/assets/a520129c-e7b9-44e2-95d0-c2c8e43d7634">
<img width="487" alt="Screenshot 2024-11-25 at 10 13 00 AM" src="https://github.com/user-attachments/assets/f642b0fe-ed73-4cef-b9eb-58cf6e839506">
<img width="307" alt="Screenshot 2024-11-25 at 10 13 10 AM" src="https://github.com/user-attachments/assets/24b358ec-613b-4103-a8f3-b2649748f862">
<img width="1412" alt="Screenshot 2024-11-25 at 10 13 20 AM" src="https://github.com/user-attachments/assets/c5d8e327-a71a-45a4-988f-fdf42155f1f7">
<img width="1186" alt="Screenshot 2024-11-25 at 10 14 14 AM" src="https://github.com/user-attachments/assets/a605d643-30aa-4f87-875a-afa153b56cb8">
